### PR TITLE
Migrate AppRepoForm to Clarity Core components

### PR DESCRIPTION
**Pull request by andresmgot**
### Description of the change

This PR migrates the current AppRepoForm (and AppRepoAddDockerCreds as subcomponent) to use Clarity React Components rather than relying on `clr-ui` styling. This is the recommended way to use Clarity with React (and fixes some of the issues from #2371). 

There are some caveats/fixes required when migrating this, I will explain them inline.

### Additional information

Before:

Screenshot from 2021-03-08 13-48-21

After (full gif):

Peek 2021-03-08 13-34

### CODE REVIEW

File: dashboard/src/components/Config/AppRepoList/AppRepoForm.tsx [hunk 1]
```diff
@@ -240,17 +243,13 @@ export function AppRepoForm(props: IAppRepoFormProps) {
...
 
   return (
-    <form onSubmit={handleInstallClick} className="app-repo-form">
-      <div className="clr-form-control">
-        <label htmlFor="kubeapps-repo-name" className="clr-control-label">
-          Name
-        </label>
-        <div className="clr-control-container">
-          <div className="clr-input-wrapper">
+    <>
```
Comment (absoludity) [inline dashboard/src/components/Config/AppRepoList/AppRepoForm.tsx:246]: Out of interest, why is this required for the migration, since `form` below is still the only single element?

**Review by absoludity — COMMENTED** [review]
It does look tedious - this is just changing the previous form elements with the cds form elements right?

**Comment by andresmgot** [comment]
> It does look tedious - this is just changing the previous form elements with the cds form elements right?

That's correct. My plan was to first do this and then add a new field for filtering packages.

File: dashboard/src/components/Config/AppRepoList/AppRepoForm.tsx (hunk 1 above)
Comment (andresmgot) [reply dashboard/src/components/Config/AppRepoList/AppRepoForm.tsx:246]: It's not required (I was just testing things).

File: dashboard/package.json [hunk 2]
```diff
@@ -84,7 +84,7 @@
     "build": "npm-run-all build-css compile-lang build-js",
     "test": "yarn run build-css && yarn run compile-lang && react-scripts test",
     "eject": "react-scripts eject",
-    "copy-clr": "shx cp ./node_modules/@clr/ui/clr-ui-dark.min.css public/clr-ui-dark.min.css && cp node_modules/@clr/ui/clr-ui.min.css public/clr-ui.min.css",
+    "copy-clr": "shx cp ./node_modules/@clr/ui/clr-ui-dark.min.css public/clr-ui-dark.min.css && cp node_modules/@clr/ui/clr-ui.min.css public/clr-ui.min.css && cp ./node_modules/@clr/ui/clr-ui-dark.min.css.map public/clr-ui-dark.min.css.map && cp ./node_modules/@clr/ui/clr-ui.min.css.map public/clr-ui.min.css.map",
```
Comment (andresmgot) [inline dashboard/package.json:87]: this couple of files (`*.css.map`) were missing and returning a 404 which caused some of the logic of Core components to crash.

File: dashboard/src/components/Config/AppRepoList/AppRepoAddDockerCreds.tsx [hunk 3]
```diff
@@ -67,123 +74,98 @@ export function AppRepoAddDockerCreds({
...
-    <div className="clr-form-columns">
-      {currentImagePullSecrets.length > 0 ? (
-        currentImagePullSecrets.map(secret => {
+    <>
+      <CdsCheckboxGroup>
+        <label>Associate Docker Registry Credentials (optional)</label>
+        {currentImagePullSecrets.length ? (
+          <CdsControlMessage>
+            Select existing secret(s) to access a private Docker registry and pull images from it.
+            More info{" "}
```
Comment (andresmgot) [inline dashboard/src/components/Config/AppRepoList/AppRepoAddDockerCreds.tsx:83]: moved this text from the AppRepoForm here since the text is actually related to this creds

File: dashboard/src/components/Config/AppRepoList/AppRepoAddDockerCreds.tsx [hunk 4]
```diff
@@ -67,123 +74,98 @@ export function AppRepoAddDockerCreds({
...
+            <input
+              id="kubeapps-docker-cred-email"
+              value={email}
+              onChange={handleEmailChange}
+              placeholder="user@example.com"
+            />
+          </CdsInput>
+          {/* TODO(andresmgot): CdsButton "type" property doesn't work, so we need to use a normal <button>
+                https://github.com/vmware/clarity/issues/5038
+                */}
```
Comment (andresmgot) [inline dashboard/src/components/Config/AppRepoList/AppRepoAddDockerCreds.tsx:167]: unfortunately this is still an issue, not sure why...

File: dashboard/src/components/Config/AppRepoList/AppRepoForm.scss [hunk 5]
```diff
@@ -1,53 +1,7 @@
...
-  .clr-checkbox-wrapper input[type="checkbox"] {
-    opacity: 1;
-  }
-  .secondary-input {
-    margin-top: 0.3rem;
-    padding: 0.6rem;
-    background-color: var(--cds-global-typography-color-100, #fff);
-  }
+.cds-textarea-fix {
+  font: caption;
```
Comment (andresmgot) [inline dashboard/src/components/Config/AppRepoList/AppRepoForm.scss:2]: this one is tricky. If not set, the clr-ui conflicts with the CdsTextarea causing that the textarea is not editable in Chrome (fun: https://github.com/vmware/clarity/issues/5687)

File: integration/use-cases/create-private-registry.js [hunk 6]
```diff
@@ -16,120 +16,120 @@ test("Creates a private registry", async () => {
 
   const randomNumber = Math.floor(Math.random() * Math.floor(100));
   const repoName = "my-repo-" + randomNumber;
-  await page.type("#kubeapps-repo-name", repoName);
+  await page.type("cds-form-group > cds-input:nth-child(1) > input", repoName);
 
   await page.type(
-    "#kubeapps-repo-url",
```
Comment (andresmgot) [inline integration/use-cases/create-private-registry.js:22]: the component ID of the inputs is automatically generated on the fly so we cannot use these fixed IDs anymore

**Review by absoludity — APPROVED** [review]
Looks great! Hope the clarity team is able to fix the issues so the migration is a bit smoother. Worth linking to the issues in the related work-arounds?
